### PR TITLE
[Android] Implement XWalkView.getUserAgentString and test case

### PR DESCRIPTION
**Pull request by xzhan96**
Implement API XWalkView.getUserAgentString for client to get its user agent.
Refine test cases for setUserAgentString and getUserAgentString.

BUG=XWALK-4817,XWALK-4918

### CODE REVIEW

**Comment by xzhan96** [comment]
@axinging @lincsoon

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @MediumTest
+    @Feature({"UserAgent"})
+    public void testUserAgentWithTestServer() throws Throwable {
+        final String customUserAgentString =
+                "testUserAgentWithTestServerUserAgent";
+
+        TestWebServer webServer = TestWebServer.start();
+        String fileName = null;
+        try {
+            final String httpPath = "/testUserAgentWithTestServer.html";
```
Comment (axinging) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java]: Did you forget this file: testUserAgentWithTestServer.html?

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Copyright (c) 2013-2014 Intel Corporation. All rights reserved.
```
Comment (axinging) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java]: For newly created file, 2015

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java (hunk 1 above)
Comment (xzhan96) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java]: I guess this should be inside Google TestWebServer..

**Comment by wuhengzhi** [comment]
This API will be deprecated when XWalkSettings was exposed?

**Comment by lincsoon** [comment]
I though it over, because there's already `setUserAgent` in `XWalkView`, let's just add `getUserAgent` too. We can let them exist together with the ones in XWalkSettings.

**Comment by wuhengzhi** [comment]
There is a file named ```SetUserAgentStringTest.java``` in xwalk core test, it could be integrated with ```UserAgentTest.java```

**Comment by xzhan96** [comment]
@axinging 
Any comments about Hengzhi's suggestion:
There is a file named SetUserAgentStringTest.java in xwalk core test, it could be integrated with UserAgentTest.java

**Comment by xzhan96** [comment]
@lincsoon @halton

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+import android.test.suitebuilder.annotation.MediumTest;
+import android.test.suitebuilder.annotation.SmallTest;
+
+import org.apache.http.Header;
+import org.apache.http.HttpRequest;
+import org.chromium.base.test.util.Feature;
+import org.chromium.net.test.util.TestWebServer;
+
+
```
Comment (halton) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java]: nit, one empty line

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        assertEquals(USER_AGENT, getUserAgent());
+        setUserAgent(null);
+        assertEquals(defaultUserAgentString, getUserAgent());
+    }
+
+    @MediumTest
+    @Feature({"UserAgent"})
+    public void testUserAgentWithTestServer() throws Throwable {
+        final String customUserAgentString =
+                "testUserAgentWithTestServerUserAgent";
```
Comment (halton) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/UserAgentTest.java]: does  squash exceed 100 chars?

**Comment by xzhan96** [comment]
@lincsoon @halton @axinging

**Comment by halton** [comment]
LGTM merging

**Pull request merged**